scripts/policy/frameworks/telemetry/log.zeek

```diff
@@ -0,0 +1,190 @@
+##! Implementation of a telemetry.log and telemetry_histogram.log file
+##! using metrics accessible via the Telemetry module.
+
+@load base/frameworks/telemetry
+
+module Telemetry;
+
+export {
+	redef enum Log::ID += { LOG, LOG_HISTOGRAM };
+
+	type Info: record {
+		## Timestamp of reporting.
+		ts: time &log;
+
+		## Peer that generated this log.
+		peer: string &log;
+
+		## Contains the values "counter" or "gauge" depending on
+		## the underlying metric type.
+		metric_type: string &log;
+
+		## The prefix (namespace) of the metric.
+		prefix: string &log;
+
+		## The name of the metric.
+		name: string &log;
+
+		## The unit of this metric, or unset if unit-less.
+		unit: string &log &optional;
+
+		## The names of the individual labels.
+		labels: vector of string &log;
+
+		## The values of the labels as listed in ``labels``.
+		label_values: vector of string &log;
+
+		## The value of this metric.
+		value: double &log;
+	};
+
+	type HistogramInfo: record {
+		## Timestamp of reporting.
+		ts: time &log;
+
+		## Peer that generated this log.
+		peer: string &log;
+
+		## The prefix (namespace) of the metric.
+		prefix: string &log;
+
+		## The name of the metric.
+		name: string &log;
+
+		## The unit of this metric, or unset if unit-less.
+		unit: string &log &optional;
+
+		## The names of the individual labels.
+		labels: vector of string &log;
+
+		## The values of the labels as listed in ``labels``.
+		label_values: vector of string &log;
+
+		## The bounds of the individual buckets
+		bounds: vector of double &log;
+
+		## The number of observations within each individual bucket.
+		values: vector of double &log;
+
+		## The sum over all observations
+		sum: double &log;
+
+		## The total number of observations.
+		observations: double &log;
+	};
+
+	## A default logging policy hook for the stream.
+	global log_policy: Log::PolicyHook;
+
+	## A default logging policy hook for the histogram stream.
+	global log_policy_histogram: Log::PolicyHook;
+
+	## Event triggered for every record in the stream.
+	global log_telemetry: event(rec: Info);
+
+	## Event triggered for every record in the histogram stream.
+	global log_telemetry_histogram: event(rec: HistogramInfo);
+
+	## How often metrics are reported.
+	option log_interval = 60sec;
```

Comment:
Suggest to move this to the top, that makes it's easier to spot option that can be changed.